internal/net/net.go

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package net
+
+import (
+	"net"
+	"strconv"
+	"time"
+)
+
+func IsLocalPortAvailable(port int) (portAvailable bool) {
+	timeout := time.Second
+	trace(
+		"Checking if local port %d is available using DialTimeout(tcp, %v, timeout: %d)",
+		port,
+		net.JoinHostPort("localhost", strconv.Itoa(port)),
+		timeout,
+	)
+	conn, err := net.DialTimeout(
+		"tcp",
+		net.JoinHostPort("localhost", strconv.Itoa(port)),
```

Comment:
>net.JoinHostPort("localhost", strconv.Itoa(port)), [](http://example.com/codeflow?start=2&length=50)

just call this once and pass it to both trace and dial

Reply:
Done (added to the follow on PR in shueybubbles/bcp)